pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -654,6 +654,14 @@ func (p *PriorityQueue) SchedulingCycle() int64 {
 func (p *PriorityQueue) determineSchedulingHintForInFlightPod(logger klog.Logger, pInfo *framework.QueuedPodInfo) queueingStrategy {
 	logger.V(5).Info("Checking events for in-flight pod", "pod", klog.KObj(pInfo.Pod), "unschedulablePlugins", pInfo.UnschedulablePlugins, "inFlightEventsSize", p.inFlightEvents.Len(), "inFlightPodsSize", len(p.inFlightPods))
 
+	failedPlugins := pInfo.UnschedulablePlugins.Union(pInfo.PendingPlugins)
+	if len(failedPlugins) == 0 {
+		// No failed plugins are associated with this Pod.
+		// Meaning something unusual (a temporal failure on kube-apiserver, etc) happened and this Pod gets moved back to the queue.
+		// In this case, we should retry scheduling it because this Pod may not be retried until the next flush.
+		return queueAfterBackoff
+	}
```

Comment:
Curious why did you move this? this check is called at line 672 anyway. Just wondering if this change is based on something specific since it is not discussed in the issue description.

Reply:
The reason is described here: https://github.com/kubernetes/kubernetes/pull/120586#discussion_r1330260118.

By the change in pkg/scheduler/schedule_one.go, if a Pod is rejected at the bind extension point, `p.inFlightPods[pInfo.Pod.UID]` would be not-found (as `Done()` removed the Pod from `p.inFlightPods` before calling the bind extension point). That'd be a problem of the original `determineSchedulingHintForInFlightPod`.

Actually, Pods rejected at the binding extension point should always go to backoffQ and we actually don't have to look up `p.inFlightPods`. 
So, here we can just reverse the order of looking up `p.inFlightPods` and checking `failedPlugins` so that such bind-failure scenario is also covered correctly.